torch/_dynamo/variables/builtin.py

```diff
@@ -1215,13 +1216,61 @@ def call_setattr(
             tx.output.side_effects.is_attribute_mutation(obj)
             and name_var.is_python_constant()
         ):
-            name = name_var.as_python_constant()
-            if name == "requires_grad" and isinstance(obj, variables.TensorVariable):
-                unimplemented(
-                    "mutating requires_grad can introduce a new leaf from non-leaf or vice versa in "
-                    "the middle of the graph, which aot_autograd does not currently know how to handle. "
-                )
-            tx.output.side_effects.store_attr(obj, name, val)
+            if isinstance(obj, variables.TensorVariable):
+                from .builder import wrap_fx_proxy
+
+                name = name_var.as_python_constant()
+                if name == "requires_grad":
+                    # TODO(voz): Make it work properly
+                    unimplemented(
```

Comment:
oh unrelated but fyi @suo (this is the "dynamo graph breaks today if you try to change the requires_grad of a tensor mid-graph)